File: cpp/s0070.cc
#include <iostream>
#include <vector>

using namespace std;

template <typename T> void print_vec(vector<T> &v) {
  cout << '[' << ' ';
  for (auto i : v)
    cout << i << ' ';
  cout << ']' << '\n';
}

class Solution {
public:
  int climbStairs(int n) {
    int v[n + 1];
    v[0] = 1;
    v[1] = 1;
    for (int i = 2; i <= n; i++) {
      v[i] = v[i - 1] + v[i - 2];
    }
    return v[n];
  }
};

int main() {
  std::vector<int> v = {};

  Solution s;

  return 0;
}

File: rust/src/s0070.rs
pub struct Solution {}

impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut result = vec![0; n as usize + 1];
        result[0] = 1;
        result[1] = 1;
        for i in 2..=n as usize {
            result[i] = result[i - 1] + result[i - 2];
        }
        result[n as usize]
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test() {
        Solution {};
        assert_eq!(Solution::climb_stairs(3), 3);
    }
}
